include/NovelRT/Exceptions/InvalidOperationException.h

```diff
@@ -0,0 +1,22 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_INVALID_OPERATION_EXCEPTION_H
+#define NOVELRT_INVALID_OPERATION_EXCEPTION_H
+
+#include <stdexcept>
+#include <string>
+
+namespace NovelRT::Exceptions {
+  class InvalidOperationException final : public std::runtime_error {
+  public:
+    InvalidOperationException() :
+      std::runtime_error("The current operation is not supported.")
```

Comment:
If its invalid, we should specify that. Not supported is a different exception. We should outline exactly what InvalidOperation means by default - it means the user attempted to perform an operation that *is* supported but is not valid/correct given the current state of the application.